resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -332,16 +335,116 @@ private InetAddress resolveHostsFileEntry(String hostname) {
         }
     }
 
+    /**
+     * Resolves the specified name into an address.
+     *
+     * @param inetHost the name to resolve
+     * @param additionals additional records ({@code OPT})
+     *
+     * @return the address as the result of the resolution
+     */
+    public final Future<InetAddress> resolve(String inetHost, Iterable<DnsRecord> additionals) {
+        return resolve(inetHost, additionals, executor().<InetAddress>newPromise());
+    }
+
+    /**
+     * Resolves the specified name into an address.
+     *
+     * @param inetHost the name to resolve
+     * @param additionals additional records ({@code OPT})
+     * @param promise the {@link Promise} which will be fulfilled when the name resolution is finished
+     *
+     * @return the address as the result of the resolution
+     */
+    public final Future<InetAddress> resolve(String inetHost, Iterable<DnsRecord> additionals,
+                                             Promise<InetAddress> promise) {
+        checkNotNull(inetHost, "inetHost");
+        checkNotNull(additionals, "additionals");
+        checkNotNull(promise, "promise");
+
+        try {
+            doResolve(inetHost, validateAdditionals(additionals), promise, resolveCache);
```

Comment:
Could we move validateAdditionals out of this try-catch block like other validations?